src/test/java/seedu/letsgethired/model/application/InternApplicationTest.java

```diff
@@ -53,6 +54,25 @@ public void isSameInternApplication() {
         assertFalse(B.isSameApplication(editedOtherInternApplication));
     }
 
+    @Test
+    public void testClone() {
+        // Create an instance of InternApplication for testing
+        InternApplication originalApplication = JANE_STREET;
+
+        // Call the clone method
+        InternApplication clonedApplication = originalApplication.clone();
+
+        // Verify that the cloned object is not the same reference
+        assertNotSame(originalApplication, clonedApplication);
+
+        // Verify that the cloned object has the same values
+        assertEquals(originalApplication.getCompany(), clonedApplication.getCompany());
```

Comment:
Perhaps we can create a helper method in `InternApplicationTest` for comparing that all the attributes of 2 different Intern Applications are the same. We can also use this helper method in `UniqueInternApplicationListTest`, which makes things easier to maintain if we ever add more fields to `InternApplication` in the future.